spec/decorator_spec.rb

```diff
@@ -0,0 +1,11 @@
+require './decorator'
+require './nameable'
+
+describe Decorator do
+  context 'Creating decorator test methods' do
+    decorator = Decorator.new('Soha Bar')
+    it 'sets decorator nameable as Soha Bar' do
+      expect(decorator.nameable).to eql 'Soha Bar'
+    end
+  end
+end
```

Comment:
- [ ] Good job making most of the changes @batoorsohail, as mentioned by the previous reviewer, we need to make sure we have tested all the methods inside all the classes, please add a test here for `correct_name` method of the `Decorator` class.